js/client-mainmenu.js

```diff
@@ -1122,12 +1122,60 @@
 					for (var i = 0; i < teams.length; i++) {
 						if ((!teams[i].format && !teamFormat) || teams[i].format === teamFormat) {
 							var selected = (i === curTeam);
-							bufs[curBuf] += '<li><button name="selectTeam" value="' + i + '"' + (selected ? ' class="sel"' : '') + '>' + BattleLog.escapeHTML(teams[i].name) + '</button></li>';
-							count++;
-							if (count % bufBoundary == 0 && curBuf < 4) curBuf++;
+							if (this.folderToggleOn !== true) {
+								bufs[curBuf] += '<li><button name="selectTeam" value="' + i + '"' + (selected ? ' class="sel"' : '') + '>' + BattleLog.escapeHTML(teams[i].name) + '</button></li>';
+								count++;
+								if (count % bufBoundary == 0 && curBuf < 4) curBuf++;
+							} else {
+								var folderName = teams[i].folder || "";
+								if (folderName !== "") {
+									if (folders[folderName] === undefined) folders[folderName] = []; // create empty array to push to
+									var thisTeam = teams[i];
+									thisTeam.id = i; // create this ID for future sorting purposes, with respect to storage
+									folders[folderName].push(thisTeam); // shows the folders with only the content of format, move out if needed
+								}
+							}
 						}
 					}
+					if (this.folderToggleOn) {
+						var keys = Object.keys(folders);
+						for (var i = 0; i < keys.length; i++) {
+							var folderData = folders[keys[i]];
+							var exists = false;
+							for (var j = 0; j < this.folderExpanded.length; j++) {
+								if (this.folderExpanded[j] === keys[i]) {
+									exists = true;
+								}
+							}
+							if (exists) {
+								for (var j = 0; j < folderData.length; j++) { // fuck i need to compare with teams
+									var selected = (folderData[j].id === curTeam);
+									bufs[curBuf] += '<li><button name="selectTeam" value="' + folders[keys[i]][j].id + '"' + (selected ? ' class="sel"' : '') + '>' + BattleLog.escapeHTML(folderData[j].name) + '</button></li>';
+									count++;
+									if (count % bufBoundary == 0 && curBuf < 4) curBuf++;
+								}
+							} else {
+								bufs[curBuf] += '<li><button name="selectFolder" class="button" value="' + keys[i] + '">' + BattleLog.escapeHTML(keys[i]) + '</button></li>';
+								count++;
+								if (count % bufBoundary == 0 && curBuf < 4) curBuf++;
+							}
+						}
+						// for (var i = 0; i < teams.length; i++) { // shows the other teams not in folders, for later changes
```

Comment:
Please don't leave in commented code. Either include it in this PR or remove and add in a followup PR.